examples/lock.ml

```diff
@@ -0,0 +1,14 @@
+type 'a holder = 'a option ref
+type 'a t = 'a holder Atomic.t
+
+let make initial = Atomic.make (ref (Some initial))
+let rec wait prev = match !prev with None -> wait prev | Some value -> value
+
+let acquire m =
+  let next = ref None in
+  (next, wait @@ Atomic.exchange m next)
+
+let release holder value =
+  assert (None == !holder);
+  holder := Some value
+  [@@inline]
```

Comment:
I run a few quick benchmarks right now and the answer seems: it depends.

Reply:
I guess it might help.  Probably depends on how aggressive the CPU is in flushing writes to memory.  On x86 there should be no difference as each write has a release fence, but on ARM that could have an effect.

I suppose the simplest thing to try is to just change the ref to an atomic and perform an `Atomic.set`.  Worth trying on ARM.

Reply:
I tried to replace the ref with an atomic and it did not seem to make noticeable difference.  A more through benchmark might expose some difference.